src/auth/webhook-secret.guard.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import * as crypto from 'crypto';
+import { Request } from 'express';
+import { Env } from 'src/env';
+
+@Injectable()
+export class WebhookSecretGuard implements CanActivate {
+  canActivate(context: ExecutionContext): boolean {
+    const request: Request = context.switchToHttp().getRequest();
+    const signature = request.headers['x-hub-signature-256'] as string;
+    const secret = Env.SECRET.WEBHOOK;
+    const payload = JSON.stringify(request.body);
+
+    if (!signature) {
+      throw new UnauthorizedException('Missing signature');
```

Comment:
PullProofAI

Ensure to handle the case where the `x-hub-signature-256` header is present but empty to avoid potential issues.
```diff
-    if (!signature) {
+    if (!signature || signature.trim() === '') {
```


<!-- This is an auto-generated comment by PullProofAI -->